includes/class-amp-theme-support.php

```diff
@@ -2107,6 +2096,15 @@ private static function get_optimizer_configuration( $args ) {
 		 */
 		$enable_ssr = apply_filters( 'amp_enable_ssr', $enable_ssr );
 
+		/**
+		 * Filter whether the AMP Optimizer should use ES modules for the runtime and extensions.
+		 *
+		 * @since 2.1.0
+		 *
+		 * @param bool $enable_esm Whether the AMP Optimizer should use ES modules for the runtime and extensions.
+		 */
+		$enable_esm = apply_filters( 'amp_enable_esm', $enable_esm );
```

Comment:
Do you think a new `amp_enable_esm` filter for this is really needed? The `amp_optimize_config` can be used to turn it off:

```php
use AmpProject\Optimizer\Transformer\RewriteAmpUrls;
use AmpProject\Optimizer\Configuration\RewriteAmpUrlsConfiguration;

add_filter( 'amp_optimizer_config', static function ( $config ) {
	$config[ RewriteAmpUrls::class ][ RewriteAmpUrlsConfiguration::ESM_MODULES_ENABLED ] = false;
	return $config;
} );
```

Reply:
No, it's not really needed, and I'm okay with removing it.

Reply:
Updated my gist to work without the filter: https://gist.github.com/schlessera/4127ab30f4235f1acbc11e3eee77701a